src/main/java/com/cognifide/aemrules/checks/ContentResourceShouldBeNullCheckedCheck.java

```diff
@@ -0,0 +1,266 @@
+        chainedMethodsNullCheck(tree);
+        super.visitMethodInvocation(tree);
+    }
+
+    @Override
+    public void visitVariable(VariableTree tree) {
+        if (isResourceInitialized(tree)) {
+            shouldBeNullChecked(tree);
+        }
+        super.visitVariable(tree);
+    }
+
+    @Override
+    public void visitClass(ClassTree tree) {
+        contentResources.clear();
+        super.visitClass(tree);
+    }
+
+    @Override
+    public void visitMethod(MethodTree tree) {
+        super.visitMethod(tree);
+        returnOccurredInsideEqualNullCheck = false;
+    }
+
+    @Override
+    public void visitReturnStatement(ReturnStatementTree tree) {
+        if (insideEqualNullCheck) {
+            returnOccurredInsideEqualNullCheck = true;
+        }
+        super.visitReturnStatement(tree);
+    }
+
+    private void shouldBeNullChecked(VariableTree tree) {
+        if (isGetContentResourceUsedOnPage(tree)) {
+            contentResources.put(tree.simpleName().name(), false);
+        }
+    }
+
+    private void cleanFlagsAfterIfStatement(IfStatementTree tree) {
+        insideIfStatement = false;
+        updateNullCheckedResources(tree, false);
+        insideEqualNullCheck = false;
+    }
+
+    private void equalsNullCheck(MethodInvocationTree tree) {
+        if (!contentResources.getOrDefault(tree.firstToken().text(), true) &&
+            !returnOccurredInsideEqualNullCheck
+        ) {
+            context.reportIssue(this, tree, RULE_MESSAGE);
+        }
+    }
+
+    private void updateNullCheckedResources(IfStatementTree tree, boolean value) {
+        if (isRightSideNullCheck(tree)) {
+            contentResources.replace(tree.condition().firstToken().text(), value);
+        } else {
+            contentResources.replace(tree.condition().lastToken().text(), value);
+        }
+    }
+
+    private void externalLibraryNullChecks(MethodInvocationTree tree) {
+        if (isNonNullUsed(tree)) {
+            contentResources.replace(tree.arguments().get(0).toString(), true);
+        } else if (isAllNonNullUsed(tree)) {
+            tree.arguments().forEach(contentResource -> contentResources.replace(contentResource.toString(), true));
+        } else if (insideIfStatement && isThisIsNullMethod(tree)) {
+            insideEqualNullCheck = true;
+        }
+    }
+
+    private void chainedMethodsNullCheck(MethodInvocationTree tree) {
+        if (tree.methodSelect() instanceof MemberSelectExpressionTree) {
+            MemberSelectExpressionTree method = (MemberSelectExpressionTree) tree.methodSelect();
+            if (method.expression() instanceof MethodInvocationTree) {
+                MethodInvocationTree invocation = (MethodInvocationTree) method.expression();
+                if (!invocation.symbol().isUnknown() &&
+                    isPage(invocation.symbol().owner().type().fullyQualifiedName()) &&
+                    isGetContentResourceUsedOnPage(invocation) && !returnOccurredInsideEqualNullCheck
```

Comment:
Move !returnOccurredInsideEqualNullCheck to new line

Reply:
done